src/mlpack/methods/ann/layer/radial_basis_function_impl.hpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file radial_basis_function_impl.hpp
+ * @author Himanshu Pathak
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "radial_basis_function.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType,
+         typename Activation>
+RBF<InputDataType, OutputDataType, Activation>::RBF() :
+    inSize(0),
+    outSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType,
+         typename Activation>
+RBF<InputDataType, OutputDataType, Activation>::RBF(
+    const size_t inSize,
+    const size_t outSize,
+    arma::mat& centres,
+    double betas) :
+    inSize(inSize),
+    outSize(outSize),
+    betas(betas),
+    centres(centres)
+{
+  sigmas = 1;
+  if (betas == 0)
+  {
+    arma::mat dis = arma::mat(outSize, centres.n_cols);
+    for (size_t i = 0; i < centres.n_cols; i++)
+    {
+      arma::mat temp = centres.each_col() - centres.col(i);
+      dis.col(i) = arma::pow(arma::sum(
+                              arma::pow((temp),
+                              2), 0), 0.5).t(); 
+    }
```

Comment:
can you do `sigmas += arma::max(... )`
and avoid using the dis matrix.

Reply:
as far as I can understand you take the maximum in each column and add it to sigmas. Is that correct?

Reply:
It should be the maximum distance between two centres I changed the implementation respectively.
Also in a test, I am using the method of multiplying by constant to get a low error we can set beta values in two ways using this implementation.